elastic/kibana
# [Lens] Fix records field name and migrate

Fixes https://github.com/elastic/kibana/issues/123824

This renames the name of the virtual "document" field to `___records___ ` - this shoudn't change the UI in any way as the display name stays i18n-ified. The weird name is chosen to make it unlikely to collide with actual user-supplied field names (`Records` wasn't a great choice for that).

To test manually:
* Create a Lens vis using the records field, save it, then switch localization (e.g. `i18n.locale: zn-CN`) - vis should still load
* Lens visualizations in sample data should work regardless in which language they were installed vs. in which they are viewed
* Old existing Lens visualizations work fine regardless of the language
